### PR TITLE
PKG-1416 rebuild gdal 3.6.2 against libxml2 2.10

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- Increase build number
- Pin libxml2 to 2.10 where applicable

### CODE REVIEW

**Review by varlackc — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ requirements:
     - libtiff
     - libuuid  # [linux]
     - libwebp-base
-    - libxml2
+    - libxml2 2.10
     - lz4-c
     - openjpeg
     - openssl
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:59]: what about other packages pinning ?

**Comment by skupr-anaconda** [comment]
Unsatisfiable dependencies on linux-64, aarch64, ppc64le, osx-64, and win64

**Comment by cbouss** [comment]
> Unsatisfiable dependencies on linux-64, aarch64, ppc64le, osx-64, and win64

Looks like libspatialite is pinned to the wrong proj version. --> Fix here: https://github.com/AnacondaRecipes/libspatialite-feedstock/pull/9

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml:59]: Since it's a rebuild I prefer keeping it as is for now.

**Pull request merged**